### PR TITLE
feat(account lib): add new fee model to transaction builder

### PR DESCRIPTION
**Pull request by micaela-ledesma**
Add fee model EIP1559 to transaction builder

Jira Ticket: https://bitgoinc.atlassian.net/browse/STLX-6729

### CODE REVIEW

File: modules/account-lib/src/coin/eth/transactionBuilder.ts [hunk 1]
```diff
@@ -400,8 +400,8 @@ export class TransactionBuilder extends BaseTransactionBuilder {
       this.validateValue(new BigNumber(fee.eip1559.maxFeePerGas));
       this.validateValue(new BigNumber(fee.eip1559.maxPriorityFeePerGas));
     }
-    if (fee.fee) {
-      this.validateValue(new BigNumber(fee.fee));
+    if (fee.gasPrice) {
```
Comment (ryan-bitgo) [inline modules/account-lib/src/coin/eth/transactionBuilder.ts:403]: shouldn't we still validate `fee.fee` ?

Comment (micaela-ledesma) [reply modules/account-lib/src/coin/eth/transactionBuilder.ts:403]: The first line of this function validates fee.fee so I modified this one to gasPrice but maybe the other one should also be in an if.

File: modules/account-lib/src/coin/eth/transactionBuilder.ts [hunk 2]
```diff
@@ -430,16 +430,23 @@ export class TransactionBuilder extends BaseTransactionBuilder {
 
   // set args that are required for all types of eth transactions
   protected buildBase(data: string): TxData {
-    return {
-      _type: ETHTransactionType.LEGACY,
+    const txData = {
```
Comment (ryan-bitgo) [inline modules/account-lib/src/coin/eth/transactionBuilder.ts:433]: change to

```
  protected buildBase(data: string): TxData {
    const baseParams = {
      gasLimit: this._fee.gasLimit,
      nonce: this._counter,
      data: data,
      chainId: this._common.chainId().toString(),
      value: this._value,
      to: this._contractAddress,
    };

    if (this._fee.eip1559) {
      return {
        ...baseParams,
        _type: ETHTransactionType.EIP1559,
        maxFeePerGas: this._fee.eip1559.maxFeePerGas,
        maxPriorityFeePerGas: this._fee.eip1559.maxPriorityFeePerGas,
      };
    } else {
      return {
        ...baseParams,
        _type: ETHTransactionType.LEGACY,
        gasPrice: this._fee?.gasPrice ?? this._fee.fee,
      };
    }
  }
```

**Review by ryan-bitgo — CHANGES_REQUESTED** [review]

File: modules/account-lib/src/coin/eth/transactionBuilder.ts (hunk 2 above)
Comment (ipelaez-simtlix) [reply modules/account-lib/src/coin/eth/transactionBuilder.ts]: Done 
image

**Review by ryan-bitgo — APPROVED** [review]

**Review by ryan-bitgo — APPROVED** [review]

**Pull request merged**